rust/src/storage/file.rs

```diff
@@ -59,4 +60,23 @@ impl StorageBackend for FileStorageBackend {
             })
         })))
     }
+
+    async fn put_obj(&self, path: &str, obj_bytes: &[u8]) -> Result<(), StorageError> {
+        let dir = std::path::Path::new(path);
+        let dir = dir.parent().unwrap();
+
+        fs::create_dir_all(dir).await.map_err(|err| StorageError::IO{ source: err })?;
```

Comment:
I think the map_err is not needed?

```suggestion
        fs::create_dir_all(dir).await?;
```